subsys/bluetooth/mesh/dfu_srv.c

```diff
@@ -451,8 +452,20 @@ static int dfu_srv_init(const struct bt_mesh_model *mod)
 		return -EINVAL;
 	}
 
+	const struct bt_mesh_model *blob_srv =
+		bt_mesh_model_find(bt_mesh_model_elem(mod), BT_MESH_MODEL_ID_BLOB_SRV);
+
+	if (blob_srv == NULL) {
+		LOG_ERR("Firmware Update Server cannot extend BLOB Transfer Server");
+		return -EINVAL;
+	}
+
 	if (IS_ENABLED(CONFIG_BT_MESH_MODEL_EXTENSIONS)) {
```

Comment:
Please remove `if (IS_ENABLED(CONFIG_BT_MESH_MODEL_EXTENSIONS))` condition for DFU server and client. I see KConfig dependencies are already correct.